### PR TITLE
Added support for Roles for Email Domain

### LINKED ISSUE #3422: Tracker for `RolesFromDomain`
This is to track the implementation of `RolesFromDomain`, which implements role setting depending on the email domain of the user.

### PR DESCRIPTION
**Pull request by pmatos**
## Contributor Checklist:

* [ X ] I have updated the unit tests
* [ X ] I have created a file in the `master/buildbot/newsfragment` directory (and read the `README.txt` in that directory)
* [ X ] I have updated the appropriate documentation

Fixes #3422

### CODE REVIEW

File: master/buildbot/www/authz/roles.py [hunk 1]
```diff
@@ -61,6 +61,26 @@ def getRolesFromUser(self, userDetails):
...
+        self.domain_roles = {}
+        for role, domains in iteritems(kwargs):
+            for domain in domains:
+                self.domain_roles.setdefault(domain, []).append(role)
+
+    def getRolesFromUser(self, userDetails):
+        if 'email' in userDetails:
+            email = userDetails['email']
+            edomain = email.split('@')[-1]
+            if edomain in self.domain_roles:
```
Comment (tardyp) [inline master/buildbot/www/authz/roles.py]: you can simplify the whole if block:

    return self.domain_roles.get(edomain, [])

**Comment by pmatos** [comment]
@tardyp thanks, now pushed new commit.

**Comment by tardyp** [comment]
@pmatos, there is still a pylint issue to fix

**Comment by pmatos** [comment]
@tardyp Will look at it now, been away during the weekend.

**Comment by pmatos** [comment]
Phew. :) All good now @tardyp 
Sorry for the re-re-re-re-reviews of the code. My environment is not really properly setup for buildbot development.

**Comment by tardyp** [comment]
thanks!

**Pull request merged**